Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -10,243 +10,424 @@
+  boost::char_separator<char> sep{"_"};
+  boost::tokenizer<boost::char_separator<char>> tokenizer{stereo_prop, sep};
+
+  Bond::BondStereo type = Bond::STEREONONE;
+  std::vector<int> atom_indexes;
+  for (const auto &t : tokenizer) {
+    if (t == "E") {
+      type = Bond::STEREOTRANS;
+    } else if (t == "Z") {
+      type = Bond::STEREOCIS;
+    } else {
+      // Atom indexes are 0-based in RDKit, and 1-based in Mae.
+      atom_indexes.push_back(FileParserUtils::toInt(t) - 1);
     }
-    AtomPDBResidueInfo *rd_info = new AtomPDBResidueInfo(
-            pdb_info.m_atom_name->at(atom_num));
+  }
+  CHECK_INVARIANT(type != Bond::STEREONONE, "bad prop value");
 
-    atom->setMonomerInfo(rd_info);
+  // We currently don't support allenes or allene-likes
+  if (atom_indexes.size() != 4) {
+    return;
+  }
 
-    if(pdb_info.m_residue_name && pdb_info.m_residue_name->isDefined(atom_num))
-    {
-        rd_info->setResidueName(pdb_info.m_residue_name->at(atom_num));
-    }
+  auto *bond = mol.getBondBetweenAtoms(atom_indexes[1], atom_indexes[2]);
+  CHECK_INVARIANT(bond, "bad stereo bond");
+  CHECK_INVARIANT(bond->getBondType() == Bond::DOUBLE, "bad stereo bond");
 
-    if(pdb_info.m_chain_id && pdb_info.m_chain_id->isDefined(atom_num))
-    {
-        rd_info->setChainId(pdb_info.m_chain_id->at(atom_num));
+  bond->setStereoAtoms(atom_indexes[0], atom_indexes[3]);
+  bond->setStereo(type);
+}
+
+//! Copy over the structure properties, including stereochemistry.
+void set_mol_properties(RWMol &mol, const mae::Block &ct_block) {
+  for (const auto &prop : ct_block.getProperties<std::string>()) {
+    if (prop.first == mae::CT_TITLE) {
+      mol.setProp(common_properties::_Name, prop.second);
+    } else if (prop.first.find(mae::CT_CHIRALITY_PROP_PREFIX) == 0 ||
+               prop.first.find(mae::CT_PSEUDOCHIRALITY_PROP_PREFIX) == 0) {
+      parseChiralityLabel(mol, prop.second);
+    } else if (prop.first.find(mae::CT_EZ_PROP_PREFIX) == 0) {
+      parseStereoBondLabel(mol, prop.second);
+    } else {
+      mol.setProp(prop.first, prop.second);
     }
+  }
+  for (const auto &prop : ct_block.getProperties<double>()) {
+    mol.setProp(prop.first, prop.second);
+  }
+  for (const auto &prop : ct_block.getProperties<int>()) {
+    mol.setProp(prop.first, prop.second);
+  }
+  for (const auto &prop : ct_block.getProperties<mae::BoolProperty>()) {
+    mol.setProp(prop.first, static_cast<bool>(prop.second));
+  }
+}
 
-    if(pdb_info.m_insertion_code &&
-            pdb_info.m_insertion_code->isDefined(atom_num))
-    {
-        rd_info->setInsertionCode(pdb_info.m_insertion_code->at(atom_num));
+//! Set atom properties. Some of these have already been parsed to construct the
+//! Atom object, and should be skipped. Also, atom properties may be undefined
+//! for an atom, and should also be skipped for that atom.
+void set_atom_properties(Atom &atom, const mae::IndexedBlock &atom_block,
```

Comment:
probably more efficient to set a single property on all atoms and then move on to the next property. We can leave that for later, though.

Reply:
I actually tried both options with just a few properties, and I didn't see too much of a difference. But, as you say, we might come back to this later on.